File: src/javascript/algorithms/array/equillibriumIndex/equilibriumIndex.js
function equilibriumIndex(nums) {
    let total = nums.reduce((a, b) => a + b, 0);
    let leftSum = 0;
    for (let i = 0; i < nums.length; i++) {
        let rightSum = total - leftSum - nums[i];
        if (leftSum === rightSum) return i;
        leftSum += nums[i];
    }
    return nums.length === 1 ? 0 : -1;
}

module.exports = equilibriumIndex;

File: src/python/algorithms/array/equillibriumIndex/equilibrium_index.py
def equilibrium_index(nums):
    total = sum(nums)
    left_sum = 0
    for i, num in enumerate(nums):
        right_sum = total - left_sum - num
        if left_sum == right_sum:
            return i
        left_sum += num
    return 0 if len(nums) == 1 else -1
